src/mame/sgi/sgi.cpp

```diff
@@ -311,6 +301,23 @@ TIMER_CALLBACK_MEMBER(sgi_mc_device::perform_dma)
 	}
 }
 
+void sgi_mc_device::update_count()
+{
+	attotime curr_time = machine().scheduler().time();
+	attotime time_delta = curr_time - m_last_update_time;
+	m_last_update_time = curr_time;
```

Comment:
You should quantise the update time to the clock frequency.

Reply:
> You should quantise the update time to the clock frequency.

I'm sorry but I didn't get your point, using host time seems more accurate in delta calculation to me?

It's a blind copy of `ioptimer.cpp` though.

Reply:
You need to quantise `curr_time` to the clock frequency to avoid accumulation of errors.  You can use `as_ticks`/`from_ticks` as a convenience.

Reply:
Done but I'm not 100% sure what I've done is what you mean.
Could you please help me double check? Thanks.